lib/datasource/npm/npmrc.ts

```diff
@@ -37,21 +34,56 @@ function envReplace(value: any, env = process.env): any {
   });
 }
 
-const envRe = regEx(/(\\*)\$\{([^}]+)\}/);
-// TODO: better add to host rules (#9588)
-function sanitize(key: string, val: string): void {
-  if (!val || envRe.test(val)) {
-    return;
+export function getMatchHostFromNpmrcHost(input: string): string {
+  if (input.startsWith('//')) {
+    const matchHost = input.replace('//', '');
+    if (matchHost.includes('/')) {
+      return 'https://' + matchHost;
+    }
+    return matchHost;
   }
-  if (key.endsWith('_authToken') || key.endsWith('_auth')) {
-    addSecretForSanitizing(val);
-  } else if (key.endsWith(':_password')) {
-    addSecretForSanitizing(val);
-    const password = fromBase64(val);
-    addSecretForSanitizing(password);
-    const username: string = npmrc[key.replace(':_password', ':username')];
-    addSecretForSanitizing(toBase64(`${username}:${password}`));
+  return input;
+}
+
+export function convertNpmrcToRules(npmrc: Record<string, any>): NpmrcRules {
+  const rules: NpmrcRules = {
+    hostRules: [],
+  };
+  const hostType = 'npm';
+  const hosts: Record<string, HostRule> = {};
+  for (const [key, value] of Object.entries(npmrc)) {
+    if (!is.string(value)) {
+      continue;
+    }
+    const keyParts = key.split(':');
+    const keyType = keyParts.pop();
+    let matchHost = '';
+    if (keyParts.length) {
+      matchHost = getMatchHostFromNpmrcHost(keyParts.join(':'));
```

Comment:
```suggestion
      matchHost = getMatchHostFromNpmrcHost(keyParts[0]);
```

Shouldn't we only pass the url and not the property name? I don't see a test for `getMatchHostFromNpmrcHost('//some.test/with/path:_auth)`

Reply:
This handles the scenario of a port in the host. See new tests